src/navsections/sectionBlueprintMiddleware/processor.ts

```diff
@@ -0,0 +1,157 @@
+import asyncLib from 'async';
+import log from 'loglevel';
+
+import {AppDispatch} from '@models/appdispatch';
+import {ItemInstance, NavigatorInstanceState, SectionInstance} from '@models/navigator';
+import {RootState} from '@models/rootstate';
+
+import {updateNavigatorInstanceState} from '@redux/reducers/navigator';
+
+import sectionBlueprintMap from '../sectionBlueprintMap';
+import {computeSectionCheckable} from './checkable';
+import {buildItemInstances} from './itemBuilder';
+import {getRowIndexToScroll, makeNavigatorRows} from './navigatorRows';
+import {computeFullScope, hasFullScopeChanged, hasSectionScopeChanged} from './scope';
+import {buildSectionInstance} from './sectionBuilder';
+import {pickPartialRecord} from './utils';
+import {computeSectionVisibility} from './visibility';
+
+/**
+ * Stores the cache of selected key:value pairs from the redux store for all section blueprints.
+ * For example, if you have "resourceMap": state.main.resourceMap in your section blueprints' getScope method,
+ * the "resourceMap" will be shallow compared only once and then added to the full scope cache.
+ * Then, all blueprints that depend on this key will be reprocessed.
+ */
+const fullScopeCache: Record<string, any> = {};
+
+/**
+ * Build the section and item instances based on the registered section blueprints
+ */
+export const processSectionBlueprints = async (state: RootState, dispatch: AppDispatch) => {
+  const previousSectionInstanceMap = state.navigator.sectionInstanceMap;
+  const previousItemInstanceMap = state.navigator.itemInstanceMap;
+
+  const sectionInstanceMap: Record<string, SectionInstance> = {};
+  const itemInstanceMap: Record<string, ItemInstance> = {};
+
+  const sectionBlueprintList = sectionBlueprintMap.getAll();
+
+  const collapsedSectionIds = state.navigator.collapsedSectionIds;
+  const sectionIdsToExpand: string[] = [];
+
+  const {fullScope, scopeKeysBySectionId, isChangedByScopeKey} = await computeFullScope({
+    state,
+    sectionBlueprintList,
+    fullScopeCache,
+  });
+
+  if (!hasFullScopeChanged({isChangedByScopeKey})) {
+    return;
+  }
+
+  await asyncLib.each(sectionBlueprintList, async sectionBlueprint => {
```

Comment:
Sure, I'm okay with removing it, just need to test it because when I initially added it I instantly saw some performance improvement. Although in this specific case maybe I could use web workers to off-load the work for each "tree"